extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -32,55 +35,113 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
+    /** @private {?string} Sent to siblings to update their position. */
+    this.y = this.element_.getAttribute('y') || '0%';
 
-    /** @private {?string} */
-    this.zoom_ = null;
+    /** @private {?string} Sent to siblings to update their position. */
+    this.zoom = this.element_.getAttribute('zoom') || '1';
+
+    /** @private {Array<Element>} */
+    this.siblings_ = [];
   }
 
   /** @override */
   buildCallback() {
-    this.x_ = this.element_.getAttribute('x') || '0%';
-    this.y_ = this.element_.getAttribute('y') || '0%';
-    this.zoom_ = this.element_.getAttribute('zoom') || '1';
+    this.getSiblings_().then((siblings_) => {
+      this.siblings_ = siblings_;
+    });
+
+    return Services.storyStoreServiceForOrNull(this.win).then(
+      (storeService) => {
+        storeService.subscribe(StateProperty.CURRENT_PAGE_ID, (currPageId) => {
+          const isOnActivePage = currPageId === this.getPageId_();
+          if (isOnActivePage) {
+            this.updateSiblings_();
+          }
+        });
+      }
+    );
   }
 
   /** @override */
   layoutCallback() {
-    const ampImgEl = dev().assertElement(
-      this.element_.querySelector('amp-img')
-    );
-    return whenUpgradedToCustomElement(ampImgEl)
-      .then(() => ampImgEl.signals().whenSignal(CommonSignals.LOAD_END))
+    return whenUpgradedToCustomElement(this.ampImgEl_)
+      .then(() => this.ampImgEl_.signals().whenSignal(CommonSignals.LOAD_END))
       .then(() => {
-        this.image_ = dev().assertElement(this.element.querySelector('img'));
+        const imgEl = dev().assertElement(this.element_.querySelector('img'));
         // Remove layout="fill" classes so image is not clipped.
-        this.image_.classList = '';
-        // Fill image to 100% height of viewport.
+        imgEl.classList = '';
+        // Centers the amp-img horizontally. The image does not load if this is done in CSS.
         // TODO(#31515): Handle base zoom of aspect ratio wider than image
-        setStyles(this.image_, {height: '100%'});
-        return this.updatePosition_();
+        setStyles(this.ampImgEl_, {
+          left: 'auto',
+          right: 'auto',
+        });
+        return Promise.resolve();
       })
       .catch(() => user().error(TAG, 'Failed to load the amp-img.'));
   }
 
   /**
+   * @return {!Promise<Array<Element>>}
+   */
+  getSiblings_() {
+    // TODO Group siblings to be transitioned together #31932
+    const siblings = Array.from(
+      document.querySelectorAll('amp-story-panning-media')
+    );
+
+    return Promise.all(
+      siblings.map((sibling) =>
+        sibling.getImpl().then((siblingImpl) => siblingImpl)
+      )
+    ).then((siblings) => Promise.resolve(siblings));
+  }
+
+  /** @private */
+  updateSiblings_() {
+    this.siblings_.forEach((siblingImpl) => {
+      // Call the updateTransform method on siblings with the active pages position values.
+      siblingImpl.updateTransform(this.x, this.y, this.zoom);
```

Comment:
You're trying to keep different siblings components in sync, which is a very common frontend problematic.
You don't need to have all the components have hard dependencies on each other and call public methods, this is the very reason why we introduced the store service.

Instead, you should store the x/y/zoom information in the store service, each component subscribes to it, and updates itself when needed.